src/cspsubscriber/DefaultCspSubscriber.ts

```diff
@@ -0,0 +1,45 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+import Logger from '../logger/Logger';
+import CspSubscriber from './CspSubscriber';
+
+export default class DefaultCspSubscriber implements CspSubscriber {
+  cspEventListenerAdded: boolean;
+  constructor(private logger: Logger) {}
+
+  destroy(): void {
+    this.unsubscribeFromCSPViolationEvent?.();
+    this.cspEventListenerAdded = false;
+  }
+
+  subscribeToCSPViolationEvent(): () => void {
```

Comment:
If we call `subscribeToCSPViolationEvent` several times in a row, the SDK fails to remove old listeners.

```
cspSubsciber.subscribeToCSPViolationEvent(); // Subscribe to "listener"
cspSubsciber.subscribeToCSPViolationEvent(); // Fail to remove the first "listener"
```

See my comment in `CspSubscriber.ts`.